spec/services/update_veteran_attr_service_spec.rb

```diff
@@ -0,0 +1,76 @@
+# frozen_string_literal: true
+
+describe UpdateVeteranAttrService do
+  let(:first_name) { "Bob" }
+  let(:last_name) { "Boberson" }
+  let(:middle_name) { "B" }
+  let(:name_suffix) { "Esq" }
+  let(:ssn) { "666000000" }
+  let(:date_of_death) { "2021-1-12" }
+  let(:veteranBob) do
+    create(
+      :veteran,
+      first_name: first_name,
+      last_name: last_name,
+      middle_name: middle_name,
+      name_suffix: name_suffix,
+      ssn: ssn,
+      date_of_death: date_of_death,
+      bgs_veteran_record: {
+        first_name: first_name,
+        last_name: last_name,
+        middle_name: middle_name,
+        name_suffix: name_suffix,
+        ssn: ssn,
+        date_of_death: date_of_death
+      }
+    )
+  end
```

Comment:
Let the factory generate the attributes, which you don't really care what the values is.
```suggestion
  let(:veteranBob) {create(:veteran)}
```